plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergNodeLocalDynamicSplitPruning.java

```diff
@@ -0,0 +1,244 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.testing.TempFile;
+import io.trino.connector.CatalogName;
+import io.trino.metadata.TableHandle;
+import io.trino.orc.OrcWriteValidation;
+import io.trino.orc.OrcWriter;
+import io.trino.orc.OrcWriterOptions;
+import io.trino.orc.OrcWriterStats;
+import io.trino.orc.OutputStreamOrcDataSink;
+import io.trino.plugin.hive.FileFormatDataSourceStats;
+import io.trino.plugin.hive.HiveConfig;
+import io.trino.plugin.hive.HiveTransactionHandle;
+import io.trino.plugin.hive.metastore.Column;
+import io.trino.plugin.hive.orc.OrcReaderConfig;
+import io.trino.plugin.hive.orc.OrcWriterConfig;
+import io.trino.plugin.hive.parquet.ParquetReaderConfig;
+import io.trino.plugin.hive.parquet.ParquetWriterConfig;
+import io.trino.spi.Page;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.connector.DynamicFilter;
+import io.trino.spi.connector.EmptyPageSource;
+import io.trino.spi.predicate.Domain;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.type.Type;
+import io.trino.testing.TestingConnectorSession;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import static io.trino.orc.metadata.CompressionKind.NONE;
+import static io.trino.orc.metadata.OrcType.createRootOrcType;
+import static io.trino.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.trino.plugin.hive.HiveType.HIVE_INT;
+import static io.trino.plugin.iceberg.IcebergFileFormat.ORC;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static io.trino.type.InternalTypeManager.TESTING_TYPE_MANAGER;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.testng.Assert.assertEquals;
+
+public class TestIcebergNodeLocalDynamicSplitPruning
+{
+    private static final String ICEBERG_CATALOG_NAME = "iceberg";
+    private static final String SCHEMA_NAME = "test";
+    private static final String TABLE_NAME = "test";
+    private static final String TABLE_SCHEMA_JSON = "" +
```

Comment:
Instead of this, can you try taking similar approach as TestIcebergSplitSource to create IcebergTableHandle ?

Reply:
Thanks for the suggestion. Indeed the creation of the table schema was much too verbose. I opted for using the iceberg `Schema` object for this purpose.

Adding the overhead of using a query runner just for retrieving the table schema seems not that good to me because the maintainer of the code might think that this is an integration test when it is a simple unit test.